File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
    
    res := [][]int{}

    if root == nil {
        return res
    }
    q := []*TreeNode{root}

    for len(q) > 0{

        level := []int{}
        n := len(q)

        for i := 0; i < n; i++{

            node := q[i]
            level = append(level, node.Val)

            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        q = q[n:]
        res = append(res, level)

        
    }

    return res
}

File: Data Structures & Algorithms/largest-bst-subtree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def recurse(self,root):
        if not root:
            return True, 0, float('inf'), float('-inf')
        
        lbst, lsize, lmin, lmax = self.recurse(root.left)
        rbst, rsize, rmin, rmax = self.recurse(root.right)

        if lbst and rbst and lmax < root.val < rmin:
            minimum = min(root.val, lmin)
            maximum = max(root.val, rmax)
            self.ans = max(self.ans, lsize + rsize + 1)
            return True, lsize + rsize + 1, minimum, maximum
        
        self.ans = max(self.ans, lsize, rsize)

        return False, 0, 0, 0


    def largestBSTSubtree(self, root: Optional[TreeNode]) -> int:
        self.ans = 0
        self.recurse(root)
        return self.ans

